Corrected various typos in contributing docs. (#15195)
django/django: docs/internals/contributing/triaging-tickets.txt

Triaging tickets
================

> Django uses Trac_ for managing the work on the codebase. Trac is a

Reviewer: I would revert; not sure that 'codebase' is a word.

Reviewer: Depends on the source ;) 

Here's a mainstream English dictionary:
<img width="432" alt="Screen Shot 2021-12-14 at 4 09 52 PM" src="https://user-images.githubusercontent.com/38668450/146079814-c4f9b3b4-30ac-4e7e-ba5e-eb61e0068641.png">

Reviewer 2: Code base / codebase is not in the style guide. Source code is though!

Author: changing it to source code then :)

community-tended garden of the bugs people have found and the features people
would like to see added. As in any garden, sometimes there are weeds to be
pulled and sometimes there are flowers and vegetables that need picking. We need
your help to sort out one from the other, and in the end, we all benefit
together.

> Like all gardens, we can aspire to perfection but in reality, there's no such

Reviewer: Comma before the coordinating "but" is more important!
```suggestion
Like all gardens, we can aspire to perfection, but in reality, there's no such
```

> thing. Even in the most pristine garden, there are still snails and insects.

Reviewer: ```suggestion
thing. Even in the most pristine garden there are still snails and insects.
```

> In a community garden, there are also helping people who -- with the best of

Reviewer: Why this edit?

Author: Either it should be "helpful person" or helping people!

Reviewer: Why can't folks be helpful? :D

Author: "helping people" resonates more I guess. :)
but I would revert if you wouldn't agree!

Reviewer: I guess I don't see a strong reason for the change. If anything it makes this garden analogy a bit worse, because the whole point is that they _aren't_ helping the garden, despite having a helpful disposition. It's about being patient with folks who actually aren't helping sometimes.

Reviewer 3: Also, we try to avoid trivial wording changes unless it's a big problem as it creates extra work for documentation translators.

> In a community garden, there are also helpful people who -- with the best of

Reviewer: ```suggestion
In a community garden there are also helpful people who -- with the best of
```

One way to help out is to *triage* tickets that have been created by other
users.

> Most of the workflow is based on the concept of a ticket's

Reviewer: I would revert.

Author: Lemme revert :p

> :ref:`triage stages <triage-stages>`. Each stage describes wherein its

Reviewer: The proposed edit is incorrect.

* Committers: people with commit access who are responsible for making the
  final decision to merge a patch.

> * Ticket triage: anyone in the Django community who chooses to

Reviewer: This is now less analogous with its surroundings; I would revert.

* Alice updates the pull request, fixing the implementation. She removes the
  "patch needs improvement" flag.

> * Daisy reviews the pull request and marks the ticket as "Ready for check-in".

Reviewer: I'm not certain if it's worth changing these if it's going to become out of sync with Trac, where it's called "checkin".

  means you could safely start writing a patch for it. This is generally more
  true for the case of accepted bugs than accepted features. A ticket for a bug
  that has been accepted means that the issue has been verified by at least one
  trigger as a legitimate bug - and should probably be fixed if possible. An

>   accepted new feature may only mean that one trigger thought the feature would

Reviewer: Edits are incorrect.

* **Accepted + Has Patch + Needs ...**

>   This means the ticket has been reviewed and has been found to need further

Reviewer: "has been found ..." is not an independent clause, so the edit is incorrect.

-----------------------

This flag means that although the ticket *has* a patch, it's not quite

> ready for check-in. This could mean the patch no longer applies

Reviewer: as above

Severity
--------

> The *severity* attribute is used to identify blockers, that is, issues that

Reviewer: 👍🏻

Keywords
--------

> With this field, you may label a ticket with multiple keywords. This can be

Reviewer: ```suggestion
With this field you may label a ticket with multiple keywords. This can be
```

> useful, for example, to group several tickets of the same theme. A Keywords can

Reviewer: "A Keywords" is not correct.

      submit support queries as tickets).

* wontfix

>       Used when a someone decides that the request isn't appropriate for

Reviewer: 👍🏻

* duplicate
      Used when another ticket covers the same issue. By closing duplicate

>       tickets, we keep all the discussions in one place, which helps

Author: ohky

      should be reopened when more information is supplied.

If you believe that the ticket was closed in error -- because you're

> still having the issue, or it's popped up somewhere else, or the triggers have

Reviewer: Triagers is correct; it refers to the people who triage tickets.

.. highlight:: console

> Regression is a bug that's present in some newer versions of Django but not in older ones. An extremely helpful piece of information is the commit that

Reviewer: Why this edit? Please revert.

Author: I think it makes sense. There are multiple older versions :D

Reviewer: Oh, I didn't even notice that part. I was looking at "Regression" instead of "A regression". "Regression is a bug" is not quite right.

Author: Changing "Regression" to "A regression" :p

> A Regression is a bug that's present in some newer versions of Django but not in older ones. An extremely helpful piece of information is the commit that

Reviewer: I wouldn't capitalize Regression, and you should wrap the line at 79 chars.

.. highlight:: console

A regression is a bug that's present in some newer versions of Django but not

> in older ones. An extremely helpful piece of information is the commit that

Reviewer: I don't see a benefit to changing this.

Reviewer 3: Me too.

Author: ```suggestion
A Regression is a bug that's present in a newer version of Django but not
in older ones. An extremely helpful piece of information is the commit that
```

Author: ```suggestion
A regression is a bug that's present in a newer version of Django but not
in older ones. An extremely helpful piece of information is the commit that
```

Reviewer: Still don't see much of a benefit, a regression can go unfixed for several versions, so "some newer versions" isn't wrong, and I would follow Mariusz's instincts here about minimizing work for translators and just fixing clear typos.

Now, we need to find a point in git history before the regression was
introduced (i.e. a point where the test passes). Use something like

> ``git checkout HEAD~100`` to check out an earlier revision (100 commits earlier,

Reviewer: 👍🏻
